R/ard_filter.R

```diff
@@ -0,0 +1,109 @@
+#' Filter Stacked Hierarchical ARDs
+#'
+#' @description `r lifecycle::badge('experimental')`\cr
+#'
+#' This function is used to filter stacked hierarchical ARDs.
+#'
+#' @param x (`card`)\cr
+#'   a stacked hierarchical ARD of class `'card'` created using [ard_stack_hierarchical()].
+#' @param filter (`expression`)\cr an expression that is used to filter rows of the hierarchical ARD. See the Details
+#'   section below for more information.
+#'
+#' @details
+#' The `filter` argument can be used to filter out rows of a hierarchical ARD which do not meet the requirements
+#' provided as an expression. Rows can be filtered on the values of any of the possible statistics (`n`, `p`, and `N`)
+#' provided they are included at least once in the ARD, as well as the values of any `by` variables. For each entry that
+#' does not meet the filtering requirement, all statistics corresponding to that entry will be removed from the ARD.
+#' Filtering is only applied to rows that correspond to the innermost variable in the hierarchy - all outer variable
+#' (summary) rows will be kept. In addition to filtering on individual statistic values, filters can be applied across
+#' the hierarchical row (i.e. across all `by` variable values) by using aggregate functions such as `sum()` and
+#' `mean()`.
+#'
+#' Some examples of possible filters:
+#' - `filter = n > 5`
+#' - `filter = n == 2 & p < 0.05`
+#' - `filter = sum(n) >= 4`
+#' - `filter = mean(n) > 4 | n > 3`
+#' - `filter = any(n > 2 & TRTA == "Xanomeline High Dose")`
+#'
+#' @return an ARD data frame of class 'card'
+#' @seealso [ard_sort()]
+#' @name ard_filter
+#'
+#' @examplesIf (identical(Sys.getenv("NOT_CRAN"), "true") || identical(Sys.getenv("IN_PKGDOWN"), "true"))
```

Comment:
I added some additional examples with text. Can you confirm what I wrote is correct?